### PR TITLE
Switch to Github Actions and remove constraints on the `cryptography` and `mysqlclient` packages.

### PR DESCRIPTION
**Pull request by symbolist**
## Description

We have a [constraint](https://github.com/edx/blockstore/blob/0424dc6/requirements/constraints.txt#L29-L36) on the `cryptography` package because the CircleCI tests use `alpine` docker image on which a Rust dependency is unavailable.

To be able to install [blockstore as a package](https://github.com/edx/blockstore/pull/97) in edx-platform, this constraint needs to be removed (edx-platform is on the latest version of the `cryptography` package).

CircleCI provides [custom docker images](https://circleci.com/docs/2.0/circleci-images/) but to be able to use Docker Hub images, to get around the rate-limiting, tokens will need to be set up (I am not clear why we haven't hit that for the `alpine` image). Instead it is simpler to switch to Github actions which is already being used for two workflows in this repo and which a number of other repos in the edx org have switched to.

## Ticket
* [BOM-2643](https://openedx.atlassian.net/browse/BOM-2643)
* [FAL-2231](https://tasks.opencraft.com/browse/FAL-2231)

## Test Instructions

* Verify the Github Actions job run the relevant tests and quality checks.
* Checkout the branch locally.
* Run
```
make destroy
make build
make dev.up
make dev.provision
make blockstore-shell

# In the shell
make test
```

## TODOs
If anything isn't yet done, list it here
- [ ] Get Github Actions CI job green.
- [ ] Remove constraints on the `cryptography` and `mysqlclient` packages.

### CODE REVIEW

**Comment by natabene** [comment]
@symbolist Thank you for your contribution. @bradenmacdonald Would you like to review this?

**Comment by bradenmacdonald** [comment]
@natabene Yes, I'd be happy to review this.

**Comment by natabene** [comment]
@bradenmacdonald Thank you!

**Comment by bradenmacdonald** [comment]
@symbolist I see this is green but still marked as a draft; let me know when it's ready for review.

**Comment by symbolist** [comment]
@bradenmacdonald It is ready for review now!

File: Dockerfile-3.8 [hunk 1]
```diff
@@ -2,12 +2,21 @@
 
 FROM ubuntu:20.04
 
-ENV VIRTUAL_ENV=/blockstore/venv
-
 RUN apt-get update
-RUN apt-get install libmysqlclient-dev libjpeg-dev libssl-dev libffi-dev python3 python3-venv python3-pip git -y
```
Comment (symbolist) [inline Dockerfile-3.8:8]: `libssl-dev` and `libffi-dev` were required by the `cryptography` package but according to the [docs](https://cryptography.io/en/latest/installation/?highlight=libffi#building-cryptography-on-linux), with the latest version of pip, a binary wheel is installed so those are no longer needed.

**Review by bradenmacdonald — APPROVED** [review]
:+1: please merge at your convenience.

- [x] I tested this: Ran `make dev.build` and `make easyserver`, confirmed GitHub build is green
- [x] I read through the code
- [x] I checked for accessibility issues: n/a
- [x] Includes documentation: n/a

**Comment by edx-webhook** [comment]
@symbolist Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/blockstore&entry.867055334=https://github.com/edx/blockstore/pull/124&entry.1484655318=https://github.com/symbolist&entry.752974735=2021-09-07+15:27&entry.1917517419=2021-09-20+14:47&entry.2133058324=No) so we can improve your experience in the future.

**Comment by edx-webhook** [comment]
Hello @symbolist: We are unable to continue with review of your submission at this time. Please see the associated JIRA ticket for more explanation.

**Comment by symbolist** [comment]
@bradenmacdonald Thanks! I think I'll have to create another PR because the current one has a CircleCI job associated with it which is preventing merge.

**Comment by symbolist** [comment]
Okay, that did not work.

@natabene I think we will need someone with Admin permissions on CircleCI to use the [Stop Building](https://app.circleci.com/settings/project/github/edx/blockstore?return-to=https%3A%2F%2Fapp.circleci.com%2Fpipelines%2Fgithub%2Fedx%2Fblockstore) option to disable CircleCI. Do you have permissions for that?

**Comment by kdmccormick** [comment]
Just disabled the build in CircleCI settings. Going to close and re-open this. fyi @natabene @symbolist

**Comment by edx-webhook** [comment]
Hello @symbolist: We are unable to continue with review of your submission at this time. Please see the associated JIRA ticket for more explanation.

**Comment by natabene** [comment]
@pomegranited Is this helpful?

**Comment by kdmccormick** [comment]
Hn, that may not have worked. I'm on a phone right now -- I can take another look in an hour when I'm on a laptop.

**Comment by natabene** [comment]
@kdmccormick Thanks for much for looking into this.

**Comment by kdmccormick** [comment]
@symbolist I amended your most recent commit (with no changes) and forced-pushed. Let's see if that works.

**Comment by kdmccormick** [comment]
Great, there we go. I think the branch's head commit hash needed to be changed in order to get rid of the CircleCI status.

@natabene @symbolist @pomegranited @bradenmacdonald This PR should be good to move forward now.

**Comment by edx-webhook** [comment]
@ormsbee, @kdmccormick: thought you might like to know that symbolist merged this pull request.

**Comment by symbolist** [comment]
@kdmccormick Thanks, that worked!

**Pull request merged**